### PR TITLE
[WIPTEST] Test openstack instance provisioning

### PR DESCRIPTION
**Pull request by rrasouli**
Purpose or Intent
=================

Moving test_provisioning to openstack repository:
Removed all irrelevant providers.
Tests included:
 -- Test provisioning from template
 -- Test provisioning from template by rest API
 -- Test provisioning  from template with additional volume (need to invetigate usage)

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit a000e2335d9f0fffb8830ccd8bb4a7c45ed4e9e5:

`cfme/tests/openstack/test_instance_provisioning.py`:
- :red_circle: Line 214:5: **N806** *variable in function should be lowercase*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit db5d67d20423ec84039cb8af6aa4228b29fc524f:

`cfme/tests/infrastructure/test_resources.py`:
- :heavy_exclamation_mark: Line 27:1: **W391** *blank line at end of file*

`cfme/tests/openstack/test_instance_provisioning.py`:
- :red_circle: Line 7:1: **F401** *'textwrap.dedent' imported but unused*
- :red_circle: Line 14:1: **F401** *'cfme.cloud.provider.gce.GCEProvider' imported but unused*
- :red_circle: Line 19:1: **F401** *'utils.update.update' imported but unused*

Please, rectify these issues :smirk: .

* 95e7f6f28981e2ea334fb70df3c6723ca1d459fc
  * Merge commit detected!!! :bangbang:
* bc1704e465b89e96566ee0cfc7a2406267ca5092
  * Merge commit detected!!! :bangbang:
* 46192b403b05be32568dca0b898fb2caf58e4706
  * Merge commit detected!!! :bangbang:
* 7131cd36f9c25ea734dcf5337f1631e95b8e6c87
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 13d9043c506395cd28039f4d6c2cdb5445f36aa3:

`cfme/tests/infrastructure/test_resources.py`:
- :heavy_exclamation_mark: Line 27:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .

* 95e7f6f28981e2ea334fb70df3c6723ca1d459fc
  * Merge commit detected!!! :bangbang:
* bc1704e465b89e96566ee0cfc7a2406267ca5092
  * Merge commit detected!!! :bangbang:
* 46192b403b05be32568dca0b898fb2caf58e4706
  * Merge commit detected!!! :bangbang:
* 7131cd36f9c25ea734dcf5337f1631e95b8e6c87
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a63dc214e2ec67a773578d54e9421c414113024c:

`cfme/tests/infrastructure/test_resources.py`:
- :heavy_exclamation_mark: Line 27:1: **W391** *blank line at end of file*

Please, rectify these issues :smirk: .

* 95e7f6f28981e2ea334fb70df3c6723ca1d459fc
  * Merge commit detected!!! :bangbang:
* bc1704e465b89e96566ee0cfc7a2406267ca5092
  * Merge commit detected!!! :bangbang:
* 46192b403b05be32568dca0b898fb2caf58e4706
  * Merge commit detected!!! :bangbang:
* 7131cd36f9c25ea734dcf5337f1631e95b8e6c87
  * Merge commit detected!!! :bangbang:

**The PR might not be acceptable in this state** :rage4:.
There is a merge commit in the branch which we do not like in PRs so please, get rid of it.

*CFME QE Bot*

File: cfme/tests/infrastructure/test_resources.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import pytest
+from utils import testgen
+
+pytestmark = [pytest.mark.meta(server_roles='+ smartproxy + smartstate')]
+
+
+def pytest_generate_tests(metafunc):
+    argnames, argvalues, idlist = \
```
Comment (RonnyPfannschmidt) [inline cfme/tests/infrastructure/test_resources.py]: this is the old style testgen, for new code use the new style

File: cfme/tests/openstack/test_instance_provisioning.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    return domain
+
+
+@pytest.fixture(scope="module")
+def cls(request, domain):
+    tcls = automate.Class(name="Methods",
+        namespace=automate.Namespace.make_path("Cloud", "VM", "Provisioning", "StateMachines",
+            parent=domain, create_on_init=True))
+    tcls.create()
+    request.addfinalizer(lambda: tcls.delete() if tcls.exists() else None)
```
Comment (RonnyPfannschmidt) [inline cfme/tests/openstack/test_instance_provisioning.py]: i believe we should introduce an `automate.Class.delete_if_exists`

**Pull request closed without merging**